### PR TITLE
Consolidate memory consts and grow operation

### PR DESCRIPTION
**Pull request by mathetake**
This commit consolidates the logic and constants around memory pages,
and memory grow, size operations. Prior to this, memory.grow and size were
implemented in each engine and embedded deeply inside of runtime, which
made it impossible for us to write unit tests on it.

### CODE REVIEW

File: wasm/store.go [hunk 1]
```diff
@@ -1842,3 +1842,26 @@ func newTableInstance(min uint32, max *uint32) *TableInstance {
 
 // UninitializedTableElementTypeID math.MaxUint64 to represent the uninitialized elements.
 var UninitializedTableElementTypeID FunctionTypeID = math.MaxUint64
+
+func (m *MemoryInstance) Grow(newPages uint32) (result uint32) {
```
Comment (mathetake) [inline wasm/store.go:1846]: will add tests on this shortly

File: RATIONALE.md [hunk 2]
```diff
@@ -13,21 +13,21 @@ Note: `microwasm` was never specified formally, and only exists in a historical
 https://github.com/bytecodealliance/wasmtime/blob/v0.29.0/crates/lightbeam/src/microwasm.rs
 
 
-### Size limitations
-#### Number of functions
```
Comment (mathetake) [inline RATIONALE.md:17]: layout was broken in #176 so I fixed it while here :D

File: wasm/mempage.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package wasm
+
+const (
+	// memoryPageSize is the unit of memory length in WebAssembly,
```
Comment (codefromthecrypt) [inline wasm/mempage.go:4]:
```suggestion
	// MemoryPageSize is the unit of memory length in WebAssembly,
```

File: wasm/mempage.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	// and is defined as 2^16 = 65536.
+	// https://www.w3.org/TR/wasm-core-1/#memory-instances%E2%91%A0
```
Comment (codefromthecrypt) [inline wasm/mempage.go:6]:
```suggestion
	// See https://www.w3.org/TR/wasm-core-1/#memory-instances%E2%91%A0
```

File: wasm/mempage.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	MemoryPageSize = 65536
+	// The maximum number of pages is defined as 2^16.
```
Comment (codefromthecrypt) [inline wasm/mempage.go:8]:
```suggestion
	// memoryMaxPages is maximum number of pages defined (2^16).
```

File: wasm/mempage.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	// https://www.w3.org/TR/wasm-core-1/#grow-mem
```
Comment (codefromthecrypt) [inline wasm/mempage.go:9]:
```suggestion
	// See https://www.w3.org/TR/wasm-core-1/#grow-mem
```

**Review by codefromthecrypt — APPROVED** [review]
sweet

**Pull request merged**